lang/cs/Org.Apache.REEF.IO/FileSystem/AzureDataLake/AzureDataLakeFileSystem.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.Azure.DataLake.Store;
+using Microsoft.Azure.Management.DataLake.Store;
+using Org.Apache.REEF.Tang.Annotations;
+
+namespace Org.Apache.REEF.IO.FileSystem.AzureDataLake
+{
+    internal sealed class AzureDataLakeFileSystem : IFileSystem
+    {
+        AdlsClient _client;
+
+        [Inject]
+        private AzureDataLakeFileSystem(AzureDataLakeStoreClient client)
+        {
+            _client = client.GetReference();
+        }
+
+        public void Copy(Uri sourceUri, Uri destinationUri)
+        {
+            throw new NotImplementedException();
+        }
+
+        public void CopyFromLocal(string localFileName, Uri remoteFileUri)
+        {
+            _client.BulkUpload(localFileName, remoteFileUri.ToString());
```

Comment:
BulkUpload is an asynchronous method but the IFileSystem interface has these methods as synchronous. You will need to implement the wait. Same goes for all the asynchronous AdlsClient methods you are using.